### PR TITLE
 Add lost taste and lost smell to the servey

### LINKED ISSUE #592: Add Loss of Taste, Loss of Smell buttons
## Summary

- [ ] Add loss of taste button to survey
- [ ] Add loss of smell button to survey
- [ ] Remove these from the suggested items in the free text question

## Motivation

These are very important symptoms to specifically track

### PR DESCRIPTION
**Pull request by pavel-ilin**
fixes #592 

### Please provide enough information so that others can review your pull request:

- Add Taste loss and Smell loss to the SurveyPage2.jsx component. Also add event handlers, data and data validation. 

- Save Taste loss and Smell loss input data to the Redux.

### Test plan (required)

Fallout the survey, 

### Final Checklist

- [x] Have you bumped the version in `package.json`?
    - Second decimal for major change, third decimal for minor change. This can go past 10 i.e. 1.0.9 !=> 1.1.0, 1.0.9 => 1.0.10
    - [Read here for more info](https://semver.org/)
- [x] Have you added any new tests necessary?
- [x] Is your PR rebased off the most current master?
- [x] Have you squashed all commits, or if you will merge will you squash all commits?
- [x] Did you use yarn, not npm?
- [x] Did you use Material-UI wherever possible?
- [x] Did you format according to Prettier?
- [x] Did you run all of your most recent changes locally to make sure everything is working?

### CODE REVIEW

**Comment by AdhamAH** [comment]
I run a fast check on it, when you select the last button it selects all the button, I tried to help by looking where the issue is but I think it will be faster if I tell you😅

File: client/src/components/survey/SurveyPage2.jsx [hunk 1]
```diff
@@ -486,6 +505,83 @@ const SurveyPage2 = props => {
           </ButtonGroup>
         </Grid>
       </Grid>
+
+      {/* BEGINING */}
```
Comment (acthelemann) [inline client/src/components/survey/SurveyPage2.jsx:509]:
```suggestion
```

File: client/src/components/survey/SurveyPage2.jsx [hunk 2]
```diff
@@ -486,6 +505,83 @@ const SurveyPage2 = props => {
...
+              className={giIssues === 'severe' ? classes.selectedButton : classes.button}
+              onClick={e => handleLostSmell(e.target.innerText)}
+            >
+              Severe
+            </Button>
+          </ButtonGroup>
+        </Grid>
+      </Grid>
+
+      {/* END */}
```
Comment (acthelemann) [inline client/src/components/survey/SurveyPage2.jsx:583]:
```suggestion
```

File: client/src/components/survey/SurveyPage2.jsx [hunk 3]
```diff
@@ -486,6 +505,83 @@ const SurveyPage2 = props => {
...
+          <Typography variant="body1" className={classes.symptomText}>
+            Loss of Taste
+          </Typography>
+        </Grid>
+        <Grid item xs={9} sm={6} className={classes.gridItem}>
+          <ButtonGroup color="secondary" aria-label="outlined primary button group" className={classes.buttonGroup}>
+            <Button
+              variant="contained"
+              value="minimal"
+              className={giIssues === 'minimal' ? classes.selectedButton : classes.button}
```
Comment (acthelemann) [inline client/src/components/survey/SurveyPage2.jsx:522]:
```suggestion
              className={lostTaste === 'minimal' ? classes.selectedButton : classes.button}
```

File: client/src/components/survey/SurveyPage2.jsx [hunk 4]
```diff
@@ -486,6 +505,83 @@ const SurveyPage2 = props => {
...
+              onClick={e => handleLostTaste(e.target.innerText)}
+            >
+              Minimal
+            </Button>
+            <Button
+              variant="contained"
+              value="moderate"
+              className={giIssues === 'moderate' ? classes.selectedButton : classes.button}
```
Comment (acthelemann) [inline client/src/components/survey/SurveyPage2.jsx:530]:
```suggestion
              className={lostTaste === 'moderate' ? classes.selectedButton : classes.button}
```

File: client/src/components/survey/SurveyPage2.jsx [hunk 5]
```diff
@@ -486,6 +505,83 @@ const SurveyPage2 = props => {
...
+              onClick={e => handleLostTaste(e.target.innerText)}
+            >
+              Moderate
+            </Button>
+            <Button
+              variant="contained"
+              value="severe"
+              className={giIssues === 'severe' ? classes.selectedButton : classes.button}
```
Comment (acthelemann) [inline client/src/components/survey/SurveyPage2.jsx:538]:
```suggestion
              className={lostTaste === 'severe' ? classes.selectedButton : classes.button}
```

File: client/src/components/survey/SurveyPage2.jsx [hunk 6]
```diff
@@ -486,6 +505,83 @@ const SurveyPage2 = props => {
...
+          <Typography variant="body1" className={classes.symptomText}>
+            Loss of Smell
+          </Typography>
+        </Grid>
+        <Grid item xs={9} sm={6} className={classes.gridItem}>
+          <ButtonGroup color="secondary" aria-label="outlined primary button group" className={classes.buttonGroup}>
+            <Button
+              variant="contained"
+              value="minimal"
+              className={giIssues === 'minimal' ? classes.selectedButton : classes.button}
```
Comment (acthelemann) [inline client/src/components/survey/SurveyPage2.jsx:558]:
```suggestion
              className={lostSmell === 'minimal' ? classes.selectedButton : classes.button}
```

File: client/src/components/survey/SurveyPage2.jsx [hunk 7]
```diff
@@ -486,6 +505,83 @@ const SurveyPage2 = props => {
...
+              onClick={e => handleLostSmell(e.target.innerText)}
+            >
+              Minimal
+            </Button>
+            <Button
+              variant="contained"
+              value="moderate"
+              className={giIssues === 'moderate' ? classes.selectedButton : classes.button}
```
Comment (acthelemann) [inline client/src/components/survey/SurveyPage2.jsx:566]:
```suggestion
              className={lostSmell === 'moderate' ? classes.selectedButton : classes.button}
```

File: client/src/components/survey/SurveyPage2.jsx [hunk 8]
```diff
@@ -486,6 +505,83 @@ const SurveyPage2 = props => {
...
+              onClick={e => handleLostSmell(e.target.innerText)}
+            >
+              Moderate
+            </Button>
+            <Button
+              variant="contained"
+              value="severe"
+              className={giIssues === 'severe' ? classes.selectedButton : classes.button}
```
Comment (acthelemann) [inline client/src/components/survey/SurveyPage2.jsx:574]:
```suggestion
              className={lostSmell === 'severe' ? classes.selectedButton : classes.button}
```

File: client/src/redux/reducers/survey.js [hunk 9]
```diff
@@ -57,6 +57,8 @@ const surveyReducer = (oldState = initialState, action) => {
             bluishnessSeverity: action.survey.bluish,
             giIssueSeverity: action.survey.giIssues,
             headacheSeverity: action.survey.headache,
+            lostTasteSeverity: action.survey.lostTasteSeverity,
+            lostSmellSeverity: action.survey.lostSmellSeverity,
```
Comment (acthelemann) [inline client/src/redux/reducers/survey.js:61]: Add this to the `TO_SURVEY_PAGE_1` case as well.

**Comment by pavel-ilin** [comment]
> I run a fast check on it, when you select the last button it selects all the button, I tried to help by looking where the issue is but I think it will be faster if I tell you😅

I can't reproduce it. After last fixes this bug is still there?  If it is we can take a look tomorrow morning together :)

**Comment by acthelemann** [comment]
I don't think your branch is up to date, can you rebase onto the current `master`?

**Comment by pavel-ilin** [comment]
Rebased.

**Pull request merged**